### PR TITLE
Update common core cache changes

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: MSAL/src/MSALAccountId.m [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- (instancetype)initWithLocalAccountIdentifier:(NSString *)identifier
+                                      objectId:(NSString *)objectId
+                                      tenantId:(NSString *)tenantId
+{
+    self = [super init];
+
+    if (self)
+    {
+        _identifier = identifier;
+        _objectId = identifier;
```
Comment (jasoncoolmax) [inline MSAL/src/MSALAccountId.m]: not `_objectId = objectId;`? :)

File: MSAL/src/MSALAccount.m [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    if (self)
+    {
+        _displayableId = [displayableId copy];
+        _name = [name copy];
+        _homeAccountId = [homeAccountId copy];
+        _localAccountId = [localAccountId copy];
+        _environment = [environment copy];
+
+        NSString *uid = clientInfo.uid;
+        NSString *utid = clientInfo.utid;
```
Comment (jasoncoolmax) [inline MSAL/src/MSALAccount.m]: Just curious, why do we need to copy a string given that its content is not changeable? :)

File: MSAL/src/MSALAccount.m [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                 clientInfo:(MSIDClientInfo *)clientInfo
+{
+    self = [super init];
+
+    if (self)
+    {
+        _displayableId = [displayableId copy];
+        _name = [name copy];
+        _homeAccountId = [homeAccountId copy];
+        _localAccountId = [localAccountId copy];
```
Comment (jasoncoolmax) [inline MSAL/src/MSALAccount.m]: Are we assigning `NSString` type to `MSALAccountId` type here? These two variables will be overwritten later at line 82/86 anyway.

File: MSAL/src/MSALAccountId.m (hunk 1 above)
Comment (oldalton) [reply MSAL/src/MSALAccountId.m]: fixed

File: MSAL/src/MSALAccount.m (hunk 2 above)
Comment (oldalton) [reply MSAL/src/MSALAccount.m]: You're correct, those are immutable strings and we don't need to copy. Fixed.

File: MSAL/src/MSALAccount.m (hunk 3 above)
Comment (oldalton) [reply MSAL/src/MSALAccount.m]: fixed, thanks

File: MSAL/src/MSALAccount+Internal.h [hunk 4]
```diff
@@ -25,61 +25,55 @@
...
  */
-@property (readonly) NSString *uid;
+@property (nonatomic) MSALAccountId *localAccountId;
 
 /*!
- Unique tenant identifier of the user. Can be nil if not returned by the service.
+ The displayable name of the account. Can be nil if not returned by the service.
  */
-@property (readonly) NSString *utid;
+@property (nonatomic) NSString *name;
```
Comment (unpluggedk) [inline MSAL/src/MSALAccount+Internal.h]: lets say displayableName here to make it clear.

File: MSAL/src/MSALAccount+Internal.h [hunk 5]
```diff
@@ -25,61 +25,55 @@
...
  @param  environment         Host part of the authority string
+ @param  tenantId            An identifier for the tenant that the account was acquired from
  */
 - (id)initWithDisplayableId:(NSString *)displayableId
                        name:(NSString *)name
-           identityProvider:(NSString *)identityProvider
-                        uid:(NSString *)uid
-                       utid:(NSString *)utid
-                environment:(NSString *)environment;
+              homeAccountId:(NSString *)homeAccountId
```
Comment (unpluggedk) [inline MSAL/src/MSALAccount+Internal.h]: Question regarding parameters above this line.
Confused between displayableId and name, as I just saw name = displayable name above.

File: MSAL/src/public/MSALAccountId.h [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+// THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+
+@interface MSALAccountId : NSObject <NSCopying>
+
+/*!
+ Unique identifier of the account in directory
+ */
+@property (nonatomic, readonly) NSString *identifier;
```
Comment (unpluggedk) [inline MSAL/src/public/MSALAccountId.h]: let's call it uniqueIdentifier if it is such to be clear?

File: MSAL/test/app/ios/MSALTestAppSettingViewController.m [hunk 7]
```diff
@@ -110,10 +110,16 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
...
-    NSIndexPath* indexPath = [NSIndexPath indexPathForRow:[self currentRow] inSection:0];
-    [_tableView selectRowAtIndexPath:indexPath
-                            animated:NO
-                      scrollPosition:UITableViewScrollPositionNone];
+
+    NSInteger currentRow = [self currentRow];
+
+    if (currentRow != -1)
+    {
+        NSIndexPath* indexPath = [NSIndexPath indexPathForRow:[self currentRow] inSection:0];
```
Comment (unpluggedk) [inline MSAL/test/app/ios/MSALTestAppSettingViewController.m]: nit: star

File: MSAL/src/MSALErrorConverter.m [hunk 8]
```diff
@@ -58,6 +58,7 @@ + (void)initialize
                                    @(MSIDErrorDeveloperAuthorityValidation) : @(MSALErrorFailedAuthorityValidation),
                                    @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
                                    @(MSIDErrorServerOauth) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorUnsupportedFunctionality): @(MSIDErrorInternal)
```
Comment (jasoncoolmax) [inline MSAL/src/MSALErrorConverter.m]: Mapping MSID error to MSID error? :)

File: MSAL/test/app/MSALStressTestHelper.m [hunk 9]
```diff
@@ -44,14 +46,13 @@ @implementation MSALStressTestHelper
 
 + (void)expireAllTokensWithClientId:(NSString *)clientId
```
Comment (jasoncoolmax) [inline MSAL/test/app/MSALStressTestHelper.m]: Seems parameter clientId is not used and tokens for all clientIds will be expired ? :)

File: MSAL/src/MSALAccount+Internal.h (hunk 5 above)
Comment (oldalton) [reply MSAL/src/MSALAccount+Internal.h]: Renamed displayableId to username to be clear. Name is based on the id_token's claim "name", so I'd prefer to just leave it that way. Please let me know if you disagree :)

File: MSAL/src/public/MSALAccountId.h (hunk 6 above)
Comment (oldalton) [reply MSAL/src/public/MSALAccountId.h]: There's actually nothing unique about it. I rephrased the comment, so it doesn't say unique.

File: MSAL/test/app/ios/MSALTestAppSettingViewController.m (hunk 7 above)
Comment (oldalton) [reply MSAL/test/app/ios/MSALTestAppSettingViewController.m]: Fixed

File: MSAL/src/MSALErrorConverter.m (hunk 8 above)
Comment (oldalton) [reply MSAL/src/MSALErrorConverter.m]: Good catch, fixed :)

File: MSAL/test/app/MSALStressTestHelper.m (hunk 9 above)
Comment (oldalton) [reply MSAL/test/app/MSALStressTestHelper.m]: Fixed

File: MSAL/src/MSALAccount+Internal.h (hunk 4 above)
Comment (oldalton) [reply MSAL/src/MSALAccount+Internal.h]: Renamed displayableId to username to be clear. Name is based on the id_token's claim "name", so I'd prefer to just leave it that way. Please let me know if you disagree :)

**Pull request merged**